### PR TITLE
✅ Continuous Integration using Travis CI & Coveralls, Model from S3, reformatted code, and badge update

### PR DESCRIPTION
**Pull request by Joshua-Alfred**
- Added necessary files for Travis CI and coveralls integration (and is working on my forked repo - and must work on this too)
- Formatted a few code lines using black
- Badge updated in README.md
- The dashboard now retrieves the model from the S3 bucket and performs the evaluation (the application runs slightly faster than before)

### CODE REVIEW

**Comment by Joshua-Alfred** [comment]
The build has failed due to the following restriction in Travis: 

https://docs.travis-ci.com/user/pull-requests/#pull-requests-and-security-restrictions

A pull request sent from a fork of the upstream repository (we call it an “external pull request”) could be manipulated to expose environment variables. The upstream repository’s maintainer would have no protection against this attack, as pull requests can be sent by anyone who forks the repository on GitHub.

Travis CI makes encrypted variables and data available only to pull requests coming from the same repository (“internal pull requests”). These are considered trustworthy, as only members with write access to the repository can send them.

Pull requests sent from forked repositories do not have access to encrypted variables or data even if these are defined in the fork source project unless certain repository settings in Travis CI aren’t set.

There is an option to "share encrypted env variables with forks (PRs)", but it is not recommended.

File: dashboard/views.py [hunk 1]
```diff
@@ -28,122 +51,111 @@ def index(request):
...
         for track in recommendations["tracks"]:
-            recommendedtracks.append({
-                'name': track['name'],
-                'artists': ', '.join([artist['name'] for artist in track['artists']]),
-                'album': track['album']['name'],
-                'uri': track['uri']
-            })
-
-        #Pass username to navbar
-        user_info = sp.current_user()
```
Comment (yp2353) [inline dashboard/views.py:65]: -

File: dashboard/views.py [hunk 2]
```diff
@@ -28,122 +51,111 @@ def index(request):
...
+            track_ids.append(track["track"]["id"])
 
-        #IF TESTING WITH TOP TRACKS INSTEAD OF RECENT
+        # IF TESTING WITH TOP TRACKS INSTEAD OF RECENT
         """ top_tracks = sp.current_user_top_tracks(limit=10, time_range='short_term')
         for track in top_tracks['items']:
             track_names.append(track['name'])
             track_artists.append(track['artists'][0]['name'])
             track_ids.append(track['id']) """
-
```
Comment (yp2353) [inline dashboard/views.py:118]: -

File: dashboard/views.py [hunk 3]
```diff
@@ -28,122 +51,111 @@ def index(request):
...
-        return JsonResponse({'result': vibe_result})
+
+        audio_features_list = sp.audio_features(track_ids)
+
+        vibe_result = check_vibe(
+            track_names, track_artists, track_ids, audio_features_list
+        )
     else:
-        # No token, redirect to login again
-        # ERROR MESSAGE HERE?
```
Comment (yp2353) [inline dashboard/views.py:133]: -

File: dashboard/views.py [hunk 4]
```diff
@@ -183,50 +212,47 @@ def extract_tracks(sp):
...
     # CHECK TRACK DATABASE BASED ON ID, ADD TRACK VIBE TO LYRICS_VIBES IF ALREADY IN DATABASE!!!
 
     lyrics_data = {}
     for track, artist, id in zip(track_names, track_artists, track_ids):
-        #SKIP TRACKS ALRDY IN DATABASE!!!
+        # SKIP TRACKS ALRDY IN DATABASE!!!
 
         query = f'"{track}" "{artist}"'
         song = genius.search_song(query)
-        if song:
```
Comment (yp2353) [inline dashboard/views.py:221]: -

File: dashboard/views.py [hunk 5]
```diff
@@ -346,39 +380,12 @@ def deduce_audio(audio_features_list):
...
 
 
 def normalize(vector):
-    #Used for testing only for now
+    # Used for testing only for now
     magnitude = np.linalg.norm(vector)
     if magnitude == 0:
         return vector
     return vector / magnitude
-
```
Comment (yp2353) [inline dashboard/views.py:359]: -

**Review by khamseaffan — APPROVED** [review]
LGTM

**Pull request merged**